sycl/test-e2e/ESIMD/unified_memory_api/Inputs/copyto_copyfrom.hpp

```diff
@@ -41,37 +41,36 @@ bool testUSM(queue Q, uint32_t Groups, uint32_t Threads,
   sycl::range<1> LocalRange{Threads};
   sycl::nd_range<1> Range{GlobalRange * LocalRange, LocalRange};
   constexpr size_t Alignment = getAlignment<T, N, false>(Properties);
+  T *In = sycl::aligned_alloc_shared<T>(Alignment, Size, Q);
   T *Out = sycl::aligned_alloc_shared<T>(Alignment, Size, Q);
   T OutVal = esimd_test::getRandomValue<T>();
-  for (int i = 0; i < Size; i++)
+  for (int i = 0; i < Size; i++) {
+    In[i] = i;
     Out[i] = OutVal;
+  }
 
   try {
     Q.submit([&](handler &cgh) {
        cgh.parallel_for(Range, [=](sycl::nd_item<1> ndi) SYCL_ESIMD_KERNEL {
          uint16_t GlobalID = ndi.get_global_id(0);
          uint32_t ElemOff = GlobalID * N;
-         simd<uint32_t, N> PassThruInt(ElemOff, 1);
-         simd<T, N> Vals = PassThruInt;
-
+         simd<T, N> Vals(ElemOff, 1);
+         simd<T, N> Input;
          if constexpr (UseProperties) {
            Vals.copy_to(Out + ElemOff, PropertiesT{});
```

Comment:
this `copy_to` cannot be verified because Out+ElemOff is rewritten few lines lower at L61

Reply:
i've added a gather in the latest commit, hopefully that will prevent optimization